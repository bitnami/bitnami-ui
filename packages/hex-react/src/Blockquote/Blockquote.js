import React from 'react';
import PropTypes from 'prop-types';
import cs from 'classnames';

const Blockquote = ({ outstanding, className, children, ...htmlProps }) => {
  const css = cs(className, { outstanding });

  return (
    <blockquote className={css} {...htmlProps}>
      {children}
    </blockquote>
  );
};

Blockquote.propTypes = {
  outstanding: PropTypes.bool,
  className: PropTypes.string,
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
  htmlProps: PropTypes.object,
};

Blockquote.defaultProps = {
  outstanding: false,
  className: '',
  children: undefined,
  htmlProps: {},
};

export default Blockquote;
