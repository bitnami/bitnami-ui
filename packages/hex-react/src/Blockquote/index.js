import BlockquoteWithHelpers from './Blockquote';

export default BlockquoteWithHelpers;
