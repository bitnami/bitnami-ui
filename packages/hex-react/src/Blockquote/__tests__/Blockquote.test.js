import React from 'react';
import { shallow } from 'enzyme';

import Blockquote from '../Blockquote';

describe('Blockquote', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Blockquote>Test</Blockquote>);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the text', () => {
    const text = 'this is a blockquote';
    const wrapper = shallow(<Blockquote>{text}</Blockquote>);
    expect(wrapper.text()).toEqual(text);
  });

  it('adds outstanding class', () => {
    const wrapper = shallow(<Blockquote outstanding />);
    expect(wrapper.hasClass('outstanding')).toBe(true);
  });

  it('adds color class', () => {
    const wrapper = shallow(<Blockquote color="brand" />);
    expect(wrapper.hasClass('brand')).toBe(true);
  });

  it('accept external classNames', () => {
    const wrapper = shallow(<Blockquote className="test" />);
    expect(wrapper.hasClass('test')).toBe(true);
  });

  it('accept any other HTML attribute', () => {
    const cite = 'https://bitnami.com/';
    const wrapper = shallow(<Blockquote cite={cite} />);
    expect(wrapper.prop('cite')).toEqual(cite);
  });
});
