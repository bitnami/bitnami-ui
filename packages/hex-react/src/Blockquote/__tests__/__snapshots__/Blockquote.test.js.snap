// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blockquote renders correctly 1`] = `
<blockquote
  className=""
  htmlProps={Object {}}
>
  Test
</blockquote>
`;
